nvm.psm1

```diff
@@ -67,30 +67,34 @@ function Set-NodeVersion {
     $nvmwPath = Get-NodeInstallLocation
 
     $requestedVersion = Join-Path $nvmwPath $VersionToUse
+    $binPath = if ([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform([System.Runtime.InteropServices.OSPlatform]::OSX)) {
+        # Under macOS, the node binary is in a bin folder
+        Join-Path $requestedVersion "bin"
+    } else {
+        $requestedVersion
+    }
 
     if (!(Test-Path -Path $requestedVersion)) {
         "Could not find node version $VersionToUse"
         return
     }
 
     # immediately add to the current powershell session path
-    $env:Path = "$requestedVersion;$env:Path"
+    # NOTE: it's important to use uppercase PATH for Unix systems as env vars
+    # are case-sensitive on Unix but case-insensitive on Windows
+    $env:PATH = $binPath + [System.IO.Path]::PathSeparator + $env:PATH
 
     if ($Persist -ne '') {
         # also add to the permanent windows path
         $persistedPaths = @($requestedVersion)
-        [Environment]::GetEnvironmentVariable('Path', $Persist) -split ';' | % {
-          if (-not($_ -like "$nvmwPath*")) {
-            $persistedPaths += $_
-          }
+        [Environment]::GetEnvironmentVariable('PATH', $Persist) -split [System.IO.Path]::PathSeparator | ForEach-Object {
+            if (-not($_ -like "$nvmwPath*")) {
+                $persistedPaths += $_
+            }
         }
-        [Environment]::SetEnvironmentVariable('Path', $persistedPaths -join ';', $Persist)
+        [Environment]::SetEnvironmentVariable('PATH', $persistedPaths -join [System.IO.Path]::PathSeparator, $Persist)
     }
 
-    $env:NODE_PATH = "$requestedVersion;"
```

Comment:
Aren't these three still required?

Reply:
Hmm seems to work on Windows still, so maybe not (I can't remember why I had them there to begin with)

Reply:
This made individual versions not isolated because the .npmrc file is shared between all versions. And afaik `NODE_PATH` is only where node_modules are looked up and I don't think it is expected behaviour to have modules resolve to global installations as a fallback. If this breaks anything let me know